src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -346,18 +346,24 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
 
         //#region automatic mining
         const nullResolved = Promise.resolve(null);
-        const mineAll = (maxTransactions: Capacity) =>
-          this.#isPaused() ? nullResolved : this.mine(maxTransactions);
+        const mineAll = (maxTransactions: Capacity, onlyOneBlock = false) =>
+          this.#isPaused()
+            ? nullResolved
+            : this.mine(maxTransactions, null, onlyOneBlock);
         if (instamine) {
           // insta mining
           // whenever the transaction pool is drained mine the txs into blocks
+          // only one transaction should be added per block
           txPool.on("drain", mineAll.bind(null, Capacity.Single));
         } else {
           // interval mining
           const wait = () =>
-            // unref, so we don't hold the chain open if nothing can interact with it
```

Comment:
`unref` would prevent promises waiting on the blockchain from actually being awaited. i.e.
```TypeScript
await new Promise(resolve => {
  blockchain.on("block", () => {
    resolve(void 0);
  });
);
```
would not be properly awaited. We already kept track of the latest timer in `this.#timer`, and we `clearTimeout(this.#timer)` when we `stop` the blockchain, so we can unref with no negative side affects.